src/rgw/rgw_rest_sts.cc

```diff
@@ -362,14 +367,16 @@ int RGWREST_STS::verify_permission()
     // If yes, then return 0, else -EPERM
     auto p_res = p.eval_principal(s->env, *s->auth.identity);
     if (p_res == rgw::IAM::Effect::Deny) {
+      ldout(s->cct, 0) << "evaluating principal returned deny" << dendl;
```

Comment:
am I correct that this just means that "authz failed" for some request? in this one message at leaset, we aren't adding any detail to the log, other than "x failed"--is that sufficient?  also, this is at level 0, which implies that this is "basic info" on what happened--does other logging already say "attempting x as user y failed"?  if so, maybe this one should be at a higher debug level?